scripts/generate_reference.py

```diff
@@ -33,9 +33,30 @@ def find_solver(solver):
         return results[0].resolve()
     raise RuntimeError("Missing solver")
 
+def find_tsgenerator(tsgenerator):
+    if sys.platform.startswith("win"):
+        suffix=".exe"
+    else:
+        suffix=""
+
+    tsgenerator_path = Path(tsgenerator)
+    if tsgenerator_path.is_file():
+        return tsgenerator.resolve()
+    if tsgenerator_path.is_dir():
+        results = []
+        for x in tsgenerator_path.iterdir():
+            if x.is_file() and (f"antares-ts-generator{suffix}" == x.name):
+                results.append(x)
+        assert(len(results) == 1)
+        return results[0].resolve()
+    raise RuntimeError("Missing ts-generator")
```

Comment:
You could add an argument taking value `solver` or `ts-generator` to avoid code duplication.